src/realm/views.cpp

```diff
@@ -310,6 +319,203 @@ DescriptorExport LimitDescriptor::export_for_handover() const
+                if (auto link_col = dynamic_cast<const LinkColumnBase*>(&col)) { // LinkColumn and ListColumn
+                    if (link_col->get_target_table() != *cur_table) {
+                        // the link does not point to the last table in the chain
+                        throw InvalidPathError(util::format("Invalid INCLUDE path at [%1, %2]: this link does not "
+                                                            "connect to the previous table ('%3').",
+                                                            i, link_ndx, cur_table->get_name()));
+                    }
+                    cur_table = link.from.get();
+                }
+                else {
+                    throw InvalidPathError(util::format("Invalid INCLUDE path at [%1, %2]: a backlink was denoted "
+                                                        "but this column ('%3') is not a link.",
+                                                        i, link_ndx, cur_table->get_column_name(link.column_ndx)));
+                }
+            }
+            else { // forward link/list
+                auto& col = tf::get_column(*cur_table, link.column_ndx);
+                columns.push_back(&col);
+                backlink_source.push_back(TableRef());
+                if (auto link_col = dynamic_cast<const LinkColumnBase*>(&col)) { // LinkColumn and ListColumn
+                    if (columns.size() == links.size()) {
+                        // An inclusion must end with a backlink column, link/list columns are included automatically
+                        throw InvalidPathError(util::format("Invalid INCLUDE path at [%1, %2]: the last part of an "
+                                                            "included path must be a backlink column.", i, link_ndx));
+                    }
+                    cur_table = &link_col->get_target_table();
+                }
+                else {
+                    // An inclusion chain must consist entirely of link/list/backlink columns
+                    throw InvalidPathError(util::format("Invalid INCLUDE path at [%1, %2]: all columns in the path "
+                                                        "must be a link/list/backlink type but this column ('%3') "
+                                                        "is a different type.",
+                                                        i, link_ndx, cur_table->get_column_name(link.column_ndx)));
+                }
+            }
+            ++link_ndx;
+        }
+    }
+}
+
+std::string IncludeDescriptor::get_description(ConstTableRef attached_table) const
+{
+    realm::util::serializer::SerialisationState basic_serialiser;
+    std::string description = "INCLUDE(";
+    for (size_t i = 0; i < m_columns.size(); ++i) {
+        const size_t chain_size = m_columns[i].size();
+        ConstTableRef cur_link_table = attached_table;
+        for (size_t j = 0; j < chain_size; ++j) {
+            if (j != 0) {
```

Comment:
auto column = chain[j];

Reply:
I'm not sure what you mean here, if it's important, can you explain a bit more?